applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp

```diff
@@ -0,0 +1,299 @@
+//
+//  Main authors:    Inigo Lopez
+//
+
+// Project includes
+#include "define_2d_wake_process.h"
+
+namespace Kratos {
+
+// Constructor for Define2DWakeProcess Process
+Define2DWakeProcess::Define2DWakeProcess(ModelPart& rBodyModelPart, const double Tolerance)
+    : Process(), mrBodyModelPart(rBodyModelPart), mTolerance(Tolerance)
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    root_model_part.CreateSubModelPart("trailing_edge_sub_model_part");
+}
+
+void Define2DWakeProcess::ExecuteInitialize()
+{
+    // Set the wake direction and normal for further computations
+    SetWakeDirectionAndNormal();
+    // Save the trailing edge for further computations
+    SaveTrailingEdgeNode();
+    // Check which elements are cut and mark them as wake
+    MarkWakeElements();
+    // Mark the elements touching the trailing edge from below as kutta
+    MarkKuttaElements();
+    // Mark the trailing edge element that is further downstream as wake
+    MartkWakeTrailingEdgeElement();
+}
+
+// This function sets the wake direction and normal for further computations
+const void Define2DWakeProcess::SetWakeDirectionAndNormal()
+{
+    // Reading the free_stream_velocity from the properties
+    const auto free_stream_velocity = mrBodyModelPart.GetProcessInfo().GetValue(FREE_STREAM_VELOCITY);
+    KRATOS_ERROR_IF(free_stream_velocity.size() != 3)
+        << "The free stream velocity should be a vector with 3 components!"
+        << std::endl;
+
+    // Computing the norm of the free_stream_velocity vector
+    const double norm = sqrt(inner_prod(free_stream_velocity, free_stream_velocity));
+
+    // The wake direction is the free stream direction
+    mWakeDirection = free_stream_velocity / norm;
+
+    // Rotating 90° to obtain the wake normal
+    mWakeNormal(0) = -mWakeDirection(1);
+    mWakeNormal(1) = mWakeDirection(0);
+    mWakeNormal(2) = 0.0;
+}
+
+// This function finds and saves the trailing edge for further computations
+const void Define2DWakeProcess::SaveTrailingEdgeNode()
+{
+    double max_x_coordinate = -1e30;
+    NodeIteratorType trailing_edge_node;
+    for (auto it_node = mrBodyModelPart.NodesBegin();
+         it_node != mrBodyModelPart.NodesEnd(); ++it_node) {
+        if (it_node->X() > max_x_coordinate) {
+            max_x_coordinate = it_node->X();
+            trailing_edge_node = it_node;
+        }
+    }
+    trailing_edge_node->SetValue(TRAILING_EDGE, true);
+    mTrailingEdgeNode = trailing_edge_node;
+}
+
+// This function checks which elements are cut by the wake and marks them as
+// wake elements
+const void Define2DWakeProcess::MarkWakeElements()
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    bool potentially_wake = false;
+    bool is_wake_element = false;
+    BoundedVector<double, 3> nodal_distances_to_wake = ZeroVector(3);
+
+    #pragma omp parallel for private(potentially_wake, is_wake_element, nodal_distances_to_wake)
```

Comment:
```suggestion
    #pragma omp parallel for firstprivate(potentially_wake, is_wake_element, nodal_distances_to_wake)
```
These are `firstprivate` vars (they are initialized before the loop).